PyHT6022/HantekFirmware/custom/device.c

```diff
@@ -29,6 +30,7 @@
 // change to support as many interfaces as you need
 BYTE altiface = 0; // alt interface
 extern volatile WORD ledcounter;
+volatile BYTE samplerate;
```

Comment:
This shouldn't be volatile.   ledcounter is only volatile, because it is changed by the timer interrupt.